### PR TITLE
sound: created a demo with dialogue prior to integrating text to speech

### LINKED ISSUE #1579: sound: research and create a demo with dialogue prior to integrating text to speech
In this issue, we will be creating a demo of an instance in the game where there will be dialogue between two npc's or a player and an npc. In order to create this demo, we will need to research and develop an understanding of the structs and implementation necessary to have a game dialogue. After we have created a demo with dialogue, we will create a new issue where we can integrate the text to speech into the demo and have it run as the game progresses.

Assignees: Rhea, Victor, Jake

### PR DESCRIPTION
**Pull request by sangerr**
In this pull request, we created a sample game with a conversation between an npc and the player. In order to understand this, we had to research the different structs necessary to create an npc and a conversation between an npc and player, which was mostly from previous npc examples and looking through their header file. After doing so, we created a demo example with a conversation that can be built upon to integrate the text to speech. 

Assignees: Rhea, Jake, Victor

### CODE REVIEW

**Review by JaStarbuck — APPROVED** [review]
This looks good! This will be good for the demo, maybe we can add some text to speech to this sample game for the main DEMO issue. It looks like the change in the README below is redundant to another issue and wasn't needed in this pull request, but afaik, it doesn't hurt anything. Good job!

File: src/sound/examples/tts_dialogue_example.c [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        printf("\n");
+        free(ret_str);
+    }
+
+    rc = -1;
+
+    printf("Friend: (again)\n");
+    while (rc != 1)
+    {
+        if (rc < 0) ret_str = start_conversation(c1, &rc, game);
```
Comment (ecoble) [inline src/sound/examples/tts_dialogue_example.c]: Please use if {...} else {...} for consistent style

**Review by ecoble — CHANGES_REQUESTED** [review]
Small style change, otherwise looks great! This will be really exciting to demo!

File: src/sound/examples/tts_dialogue_example.c (hunk 1 above)
Comment (sangerr) [reply src/sound/examples/tts_dialogue_example.c]: I just pushed an updated version where I fixed this issue, I had an issue in that I didn't git pull on this branch before uploading, but I was able to merge the two versions and the changes can be viewed under "fixed if else formatting and other typos" commit.

**Review by ecoble — APPROVED** [review]
Looks good!

**Comment by jwschroeder21** [comment]
Summary: 

We created a sample game with a conversation between an npc and the player. We researched the different structs necessary to create an npc and a conversation between an npc and player. We then created a demo example with a conversation that can be built upon to integrate the text to speech. There were some initial stylistic issues with if else statements, which we fixed under the commit "fixed if else formatting and other typos" commit. With Emily and Jason's approval, this pull request was successfully merged!

**Comment by ecoble** [comment]
Issue Score: Excellent

Comments:
Great job creating this demo, and also incorporating integration with NPCs!

**Pull request merged**